File: 0075-sort-colors/0075-sort-colors.dart
class Solution {
  void sortColors(List<int> nums) {
    
    bool swapped;
    int temp;

    for(int i = 0; i < nums.length - 1; i++)
    {
        swapped = false;

        for(int j = 0; j < nums.length - 1 - i; j++)
        {
            if(nums[j] > nums[j + 1])
            {
                temp = nums[j];
                nums[j] = nums[j + 1];
                nums[j + 1] = temp;
                swapped = true;
            }
        }
        if(!swapped) 
            break;
    }
  }
}

File: 0075-sort-colors/0075-sort-colors.kt
class Solution {
    fun sortColors(nums: IntArray): Unit {

        for(i in 0..nums.size - 1) {
            var isSorted = true
            for(j in 1..nums.size - i - 1) {
                if(nums[j - 1] > nums[j]) {
                    var temp = nums[j]
                    nums[j] = nums[j - 1]
                    nums[j - 1] = temp
                    isSorted = false
                }
            } 
            if(isSorted) return
        }
    }
}